content/blog/registry-writing-i2c-component/index.md

```diff
@@ -0,0 +1,742 @@
+
+The ESP32-C3-DevKit-RUST-1 development board has 2 sensors connected to the I2C peripheral on the following GPIOs:
+
+| Signal     | GPIO        |
+|------------|-------------|
+| SDA        | GPIO10      |
+| SCL        | GPIO8       |
+
+All the project files are available on the project GitHub.
+
+{{< github repo="esp-rs/esp-rust-board" >}}
+
+### The new ESP-IDF I2C driver
+
+From the ESP-IDF version 5.2, the I2C driver has been redesigned and we recommend to use this new driver instead of the legacy. Espressif provides the [migration guide](https://docs.espressif.com/projects/esp-idf/en/release-v5.2/esp32/migration-guides/release-5.x/5.2/peripherals.html#i2c) and the [I2C peripheral documentation](https://docs.espressif.com/projects/esp-idf/en/release-v5.2/esp32/api-reference/peripherals/i2c.html) for reference.
+
+The new features for this new driver includes:
+
+- New initialization mode.
+- Thread safety.
+- The APIs were simplified and some were removed.
+
+Please not if you are still using the legacy driver, update to the new driver since the legacy will be removed in the future.
+
+### Creating the component
+
+The Component Manager [documentation](https://docs.espressif.com/projects/idf-component-manager/en/latest/) is very comprehensive and you will find all the information you need there, including the [simple component](https://docs.espressif.com/projects/idf-component-manager/en/latest/guides/packaging_components.html#a-simple-esp-idf-component) structure.
+
+To create the component, we will go through some steps:
+
+1. Create the new component using the command line.
+2. Create the component manifest file and set with the component information.
+3. Set the license.
+4. Write the README.md file.
+5. Write the code for the sensor driver.
+6. Write the example.
+
+We assume that you already have the ESP-IDF installed on your system. If you don't, please check our article about installing the ESP-IDF.
+
+On the Espressif DevCon23, Ivan Grokhotkov, gave a talk titled: Developing, Publishing, and Maintaining Components for ESP-IDF. You can watch this talk as an additional material for your studies.
+
+{{< youtube D86gQ4knUnc >}}
+
+#### Create the new component
+
+The process will be done by the CLI (Command-line Interface) tool `idf.py`.
+
+```bash
+idf.py create-component shtc3
+```
+
+After that, the new folder should contain the following structure:
+
+```text
+.
+└── shtc3
+    ├── CMakeLists.txt
+    ├── include
+    │   └── shtc3.h
+    └── shtc3.c
+```
+
+#### Create the Manifest file
+
+The manifest file is mandatory and this is the file that the Registry will recognize it as a component. The bare-minimum for this file is described in the documentation, but we will fill with the most common important fields.
+
+Create the `idf_component.yml` in the component root directory with this content. You can change it if you want.
+
+```yaml
+version: 1.0.0
+targets:

```

Comment:
While it's worth mentioning how can one list a specific target, for universal componet like this it might make sense to skip all the lines, which will be future proof and component could work with future chips introduced in ESP-IDF. I guess in this type of sensor, there is nothing specific which would require just specific family.